### ISSUE TITLE
Japanese Font Issue

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#6788]
Originally post by @rachelmcr: 
A Japanese iOS user reports that there’s a problem with the editor and preview font for Japanese
> While previewing and editing posts in WordPress app using Japanese, the fonts are represented in a Chinese version of the font, not Japanese. They look the same from an unknowing eye, but completely different fonts.

![japanese-iospreview](https://cloud.githubusercontent.com/assets/1435271/23411403/ce24102a-fd97-11e6-8276-a826769e91c0.jpeg)

The issues is not immediately obvious to the untrained eye, and Naoko helpfully provided some context and highlighted the characters that are incorrectly represented:
<img width="898" alt="character-differences" src="https://cloud.githubusercontent.com/assets/1435271/23411511/1ff4c64c-fd98-11e6-87c9-2136a64ad070.png">

She also found [an article](https://href.li/?http://qiita.com/usagimaru/items/46c27c42e862fb8aba7e#uifontdescriptor) providing some background on the issue and potential fix.

TL;DR:
The font phenomenon is due to the way iOS prioritizes language settings, and can occur when other languages are listed under Settings > General > Language & Region > Preferred Language Order and ordered before Japanese. In particular when Chinese is listed before Japanese.  
Per Naoko: `However, the article says by default Chinese has priority over Japanese. So even if the user doesn’t have Chinese in the “Preferred language order”, this could happen (unless Japanese is in the order list).`
<img width="376" alt="general-language-pref" src="https://cloud.githubusercontent.com/assets/1435271/23411586/57ab60aa-fd98-11e6-8117-4e2f1914bf4e.png">

The issue can be resolved by changing the order so Japanese is prioritized over Chinese.

The issue can also be resolved programmatically as well, but I’ll need to give that part a more careful read. My understanding is the author describes two different approaches, one that uses a sort of mixed font by creating a custom UIFontDescriptor (but is an imperfect solution), and another approach that uses the poorly documented kCTLanguageAttributeName NSAttributeString attribute to force set the correct system language for the text.

Per Naoko: `What I read from the article is that setting NSAttributedString with an attribute kCTLanguageAttributeName = ja is the best solution`

**Comment by rachelmcr** [comment]
Another user sent in feedback reporting this behavior. Their translated message:

> I use it in Japanese, but Chinese fonts are used for kanji and it is hard to read. Please improve to use Japanese fonts.

Ref: AppbotX 27235

**Comment by rachelsquirrel** [comment]
I've got a similar report in #199509-h but it's affecting tags.
